### DISCUSSION TITLE
Will there be a change of syntax because of V9

### DISCUSSION THREAD

**Discussion by wortkotze** [invertase/react-native-firebase#6161] [Q&A] [answered]
Hi,

I need your help. We swapped our app to firebase-sdk v9 but recognized now that we are not able to use appcheck etc because it is using the normal firebase websdk.

We would now swap to react-native-firebase will there be a change of the sdk to the new syntax of v9 in the near future or will this lib stay in the v8 way of working with firebase?

if it will change to v9 is there already something on the roadmap

thx
Stephan

**Comment by mikehardy** [comment, marked as answer]
We will eventually switch to the new v9 APIs yes, but it is not under active development at the moment
I'm still using the v8 JS SDK items as I do react-native-web projects personally, for this reason, as I shim in react-native-firebase depending on whether it's web or not

https://github.com/invertase/react-native-firebase-authentication-example/tree/main/template/src/shims
https://github.com/invertase/react-native-firebase-authentication-example/blob/68bd37f8c219f892655b40c73e9818370f368b82/template/craco.config.js#L111-L114
